app/models.py

```diff
@@ -1,105 +1,160 @@
 """
 This files handles all the database logic and instances
 """
-import itertools
+from datetime import datetime, timedelta
+import jwt
 from passlib.apps import custom_app_context as pwd_context
 
-from app import db
+from app import db, app
 
-class Users:
+#association table that associates userstable to events table
+rsvp = db.Table('rsvps',
+                db.Column('user_id', db.Integer, db.ForeignKey('user_db.id')),
+                db.Column('event_id', db.Integer,
+                          db.ForeignKey('events_db.id'))
+               )
+class Users(db.Model):
     """
     This class handles all the logic and methods
     associated with a user
     """
-    user_db = {}
-    id_generator = itertools.count(1)
+
+    __tablename__ = 'user_db'
+
+    id = db.Column(db.Integer, primary_key=True)
+    email = db.Column(db.String(256), nullable=False, unique=True)
+    password = db.Column(db.String(256), nullable=False)
+    fname = db.Column(db.String(256), nullable=False)
+    lname = db.Column(db.String(256), nullable=False)
+    event = db.relationship(
+        'Events', order_by='Events.id', cascade="all, delete-orphan"
+    )
+    event_rsvp = db.relationship(
+        'Events', secondary=rsvp, backref=db.backref('rsvp', lazy='dynamic')
+    )
 
     def __init__(self, fname, lname, email, password):
-        self._id = next(self.id_generator)
         self.fname = fname
         self.lname = lname
         self.email = email
-        self.password = password
+        self.password = pwd_context.encrypt(password)
 
     def save(self):
-        """Saves the data to the datastructure dictonary"""
-        self.user_db[self._id] = dict(
-            fname=self.fname,
-            lname=self.lname,
-            email=self.email,
-            password=self.password
-        )
-        self.user_db[self.email] = self.password
-        return self
-
-    def check_user(self, email):
-        """
-        This method takes in a email and
-        checks if its in the dictonary
-        """
-        if email in self.user_db.keys():
-            return "User already exists. Please login.", 201
-        return False
+        """Creates a new user and saves to the database"""
+        db.session.add(self)
+        db.session.commit()
 
-    def hash_password(self, password):
+    @staticmethod
+    def check_user(email):
         """
-        hash pasword to store in db
+        This method takes in a email and
+        checks if its in the database
         """
-
-        self.password = pwd_context.encrypt(password)
+        return Users.query.filter_by(email=email).first()
 
     def verify_password(self, password):
         """
         check pasword provided with hash in db
         """
-        pass
-        # return pwd_context.verify(password, self.password)
-
+        return pwd_context.verify(password, self.password)
+
+    def generate_token(self, user_id):
```

Comment:
good implementation of the generate token.
5 min for a token to expire won't that be too little for a client?

Reply:
try and squash your commit so that someone can know what to review